src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -886,6 +892,47 @@ private void StartMusic()
         Jukebox.Instance.Resume();
     }
 
+    /// <summary>
+    ///   Updates the arrowPosition variable to the top most point of the middle 3 rows
+    ///   Should be called on any layout change
+    /// </summary>
+    private void UpdateArrow()
+    {
+        // 1. Retrieves all the hexes in the microbe
+        // 2. Fetches only the middle 3 rows
+        // 3. Maps the R position to the actual height
+        // 4. Retrieves the minimum value, which is the top most location in the middle 3 rows
+        // The calculation falls back to 0 if there are no hexes found in the middle 3 rows
+        var highestPointInMiddleRows = 0.0f;
+
+        foreach (var organelle in editedMicrobeOrganelles)
+        {
+            foreach (var relativeHex in organelle.Definition.Hexes)
+            {
+                var absoluteHex = relativeHex + organelle.Position;
+
+                if (absoluteHex.Q < -1 || absoluteHex.Q > 1)
+                    continue;
+
+#pragma warning disable 8509 // nothing else possible due to the if above
+                var r = absoluteHex.Q switch
+#pragma warning restore 8509
+                {
+                    -1 => absoluteHex.R - 1,
+                    0 => absoluteHex.R - .5f,
+                    1 => absoluteHex.R,
+                };
+
+                highestPointInMiddleRows = Mathf.Min(highestPointInMiddleRows, r);
```

Comment:
Hang on, what does this math do? Shouldn't you just call the method to convert the R and Q style coordinates to cartesian here?

Reply:
The method is in the Hex class, and is called: AxialToCartesian